honeybee_energy/writer.py

```diff
@@ -22,44 +28,368 @@ def generate_idf_string(object_type, values, comments=None):
         ep_str = object_type + ',\n ' + '\n '.join(
             '{},{}!- {}'.format(val, space, com) for val, space, com in
             zip(values[:-1], spaces[:-1], comments[:-1]))
-        ep_str = ep_str + '\n {};{}!- {}'.format(
-            values[-1], spaces[-1], comments[-1])
+        if len(values) == 1:
+            ep_str = ep_str + '{};{}!- {}'.format(values[-1], spaces[-1], comments[-1])
+        else:
+            ep_str = ep_str + '\n {};{}!- {}'.format(values[-1], spaces[-1], comments[-1]) \
+                if comments[-1] != '' else ep_str + '\n {};'.format(values[-1])
     else:
         ep_str = object_type + ',\n ' + '\n '.join(
             '{},'.format(val) for val in values[:-1])
-        ep_str = ep_str + '\n {};'.format(values[-1])
+        ep_str = ep_str + '\n {};'.format(values[-1]) if len(values) == 1 else \
+            ep_str + '{};'.format(values[-1])
     return ep_str
 
 
+def door_to_idf(door):
+    """Generate an IDF string representation of a Door.
+
+    Note that the resulting string does not include full construction definitions.
+
+    Args:
+        door: A honeyee Door for which an IDF representation will be returned.
+    """
+    door_bc_obj = door.boundary_condition.boundary_condition_object if \
+        isinstance(door.boundary_condition, Surface) else ''
+    values = (door.name,
+              'Door' if not door.is_glass else 'GlassDoor',
+              door.properties.energy.construction.name,
+              door.parent.name if door.has_parent else 'unknown',
+              door_bc_obj,
+              door.boundary_condition.view_factor,
+              '',  # TODO: Implement Frame and Divider objects on WindowConstructions
+              '1',
+              len(door.vertices),
+              ',\n '.join('%.3f, %.3f, %.3f' % (v[0], v[1], v[2])
```

Comment:
What about unpacking v?

```suggestion
              ',\n '.join('%.3f, %.3f, %.3f' % (*v)
```

Reply:
I see this in a couple of places but it is not really critical.

Reply:
I agree what you have there is much cleaner.